common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -155,22 +158,66 @@ private boolean handleUrl(final WebView view, final String url) {
             return true;
         } else if (isRedirectUrl(formattedURL)) {
             Logger.info(TAG, "Navigation starts with the redirect uri.");
-            return processRedirectUrl(view, url);
+            processRedirectUrl(view, url);
+            return true;
         } else if (isWebsiteRequestUrl(formattedURL)) {
             Logger.info(TAG, "It is an external website request");
-            return processWebsiteRequest(view, url);
+            processWebsiteRequest(view, url);
+            return true;
         } else if (isInstallRequestUrl(formattedURL)) {
             Logger.info(TAG, "It is an install request");
-            return processInstallRequest(view, url);
+            processInstallRequest(view, url);
+            return true;
         } else if (isWebCpUrl(formattedURL)) {
             Logger.info(TAG, "It is a request from WebCP");
-            return processWebCpRequest(view, url);
+            processWebCpRequest(view, url);
+            return true;
+        } else if (isPlayStoreUrl(formattedURL)){
+            Logger.info(TAG, "Request to open PlayStore.");
+            return processPlayStoreURL(view, url);
+        } else if (isAuthAppMFAUrl(formattedURL)){
+            Logger.info(TAG, "Request to link account with Authenticator.");
+            processAuthAppMFAUrl(url);
+            return true;
+        } else if (isValidRedirectUri(url)){
+            Logger.info(TAG, "Check for Redirect Uri.");
+            processInvalidRedirectUri(view, url);
+            return true;
+        } else if (isBlankPageRequest(formattedURL)){
+            Logger.verbose(TAG, "It is an blank page request");
+            return true;
+        } else if (isUriSSLProtected(formattedURL)){
+            Logger.verbose(TAG, "Check for SSL protection");
+            processSSLProtectionCheck(view, url);
+            return true;
         } else {
-            Logger.info(TAG, "It is an invalid redirect uri.");
-            return processInvalidUrl(view, url);
+            Logger.info(TAG, "This maybe a valid URI, but no special handling for this mentioned URI, hence deferring to WebView for loading.");
+            processInvalidUrl(url);
+            return false;
         }
```

Comment:
nit: you can also return true here in the end.. and remove it from the else blocks above.